web/src/user_profile.js

```diff
@@ -275,6 +331,21 @@ function handle_remove_stream_subscription(target_user_id, sub, success, failure
     }
 }
 
+function handle_add_stream_subscription(target_user_id, sub, success, failure) {
+    if (people.is_my_user_id(target_user_id)) {
+        // Self subscribe.
+        const color = sub.color;
+        channel.post({
+            url: "/json/users/me/subscriptions",
+            data: {subscriptions: JSON.stringify([{name: sub.name, color}])},
+            success,
+            error: failure,
+        });
+    } else {
+        subscriber_api.add_user_ids_to_stream([target_user_id], sub, success, failure);
```

Comment:
I think it'd probably be worth inlining this logic into `subscriber_api.add_current_user_to_stream` function rather than duplicating it here -- if the `user_ids` list is `[people.my_current_user_id()`, then do this?

Reply:
(And then you wouldn't need this function, the caller could just call `subscriber_api.add_user_ids_to_stream`

Reply:
Sure, changed it